source/client-side-operations-timeout/tests/runCursorCommand.yml

```
@@ -0,0 +1,316 @@
+      id: &commandDb commandDb
+      client: *commandClient
+      databaseName: *commandDb
+  - database:
+      id: &db db
+      client: *client
+      databaseName: *db
+  - collection:
+      id: &collection collection
+      database: *db
+      collectionName: *collection
+
+initialData:
+  - collectionName: *collection
+    databaseName: *db
+    documents: &documents
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+      - { _id: 4, x: 44 }
+      - { _id: 5, x: 55 }
+
+tests:
+  - description: errors if timeoutMode is set without timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMode: cursorLifetime
+        expectError:
+          isClientError: true
+
+  - description: error if timeoutMode is cursorLifetime and cursorType is tailableAwait
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMode: cursorLifetime
+          cursorType: tailableAwait
+        expectError:
+          isClientError: true
+
+  - description: error if getMore maxTimeMS setting is greater than timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMS: 5
+          maxTimeMS: 10
+        expectError:
+          isClientError: true
+
+  - description: error if getMore maxTimeMS setting is equal to timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMS: 5
+          maxTimeMS: 5
+        expectError:
+          isClientError: true
+
+  # If timeoutMode is unset, it should default to CURSOR_LIFETIME and the time remaining after the find succeeds should be applied to the getMore
+  - description: Non-tailable cursor lifetime remaining timeoutMS applied to getMore if timeoutMode is unset
+    operations:
+      # Block find/getMore for 15ms.
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+              failCommands: [find, getMore]
+              blockConnection: true
+              blockTimeMS: 15
+      # Run a find with timeoutMS=20 and batchSize=1 to force two batches, which will cause a find and a getMore to be sent.
+      # Both will block for 15ms so together they will go over the timeout.
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          timeoutMS: 20
+          command: { find: *collection, batchSize: 2 }
```

```suggestion
          command: { find: *collection, batchSize: 1 }
```

Should this be a "1"?